### PR TITLE
WRKLDS-1492: Update k8s dependencies to 1.31.1

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR updates k8s dependencies to 1.31.1 in addition to the required OCP libraries.

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
I'd like to check pod logs first to see everything is ok

**Comment by ardaguclu** [comment]
After checking the CI logs, cluster-openshift-controller-manager-operator seems to be healthy and no issue
/hold cancel

File: pkg/operator/generic_client.go [hunk 1]
```diff
@@ -87,3 +95,122 @@ func (p *genericClient) UpdateOperatorStatus(ctx context.Context, resourceVersio
...
+	if desiredConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	instance, err := p.client.OpenShiftControllerManagers().Get(ctx, "cluster", metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		// do nothing and proceed with the apply
+		v1helpers.SetApplyConditionsLastTransitionTime(p.clock, &desiredConfiguration.Conditions, nil)
+		desiredStatus := &v1.OpenShiftControllerManagerStatusApplyConfiguration{
```
Comment (ingvagabund) [inline pkg/operator/generic_client.go:147]: This and the default block have some code duplication. Worth de-duplicating after the switch.

Comment (ardaguclu) [reply pkg/operator/generic_client.go:148]: Deduplicated some code portions. Thanks.

Comment (ardaguclu) [reply pkg/operator/generic_client.go:148]: @ingvagabund CI passed. Could you PTAL one more round?.

File: pkg/operator/generic_client.go [hunk 2]
```diff
@@ -147,7 +148,6 @@ func (p *genericClient) ApplyOperatorStatus(ctx context.Context, fieldManager st
 		desiredStatus := &v1.OpenShiftControllerManagerStatusApplyConfiguration{
 			OperatorStatusApplyConfiguration: *desiredConfiguration,
 		}
-		desired := v1.OpenShiftControllerManager("cluster")
 		desired.WithStatus(desiredStatus)
 		_, err = p.client.OpenShiftControllerManagers().ApplyStatus(ctx, desired, metav1.ApplyOptions{
```
Comment (ingvagabund) [inline pkg/operator/generic_client.go:152]: This line is duplicated as well

Comment (ardaguclu) [reply pkg/operator/generic_client.go]: yeah, good catch. I forgot to remove this and if you didn't catch, we'd execute ApplyStatus twice. Thank you.

**Pull request merged**